source/nut/usr/local/emhttp/plugins/nut/include/nut_status.php

```diff
@@ -29,6 +29,18 @@
 
 if (file_exists('/var/run/nut/upsmon.pid')) {
   exec("/usr/bin/upsc ".escapeshellarg($nut_name)."@$nut_ip 2>/dev/null", $rows);
+
+  if ($_GET['diagsave'] == "true") {
+
```

Comment:
I think it could be fine to anonymize diagnostics by removing serial number from `device.serial` in the dump to avoid any related privacy issue.

This code should be enough:
```php
array_walk($rows, function(&$var) {
  if (preg_match('/^device.serial:/i', $var))
    $var = 'device.serial: [REMOVED]';
});
```

Reply:
I noticed that there is also `ups.serial`.

Update:
```php
array_walk($rows, function(&$var) {
  if (preg_match('/^(device|ups).serial:/i', $var, $matches))
    $var = $matches[1] . '.serial: [REMOVED]';
});
```

Reply:
thanks a lot, implemented